### PR TITLE
Added windows toolchain

### PR DESCRIPTION
**Pull request by AKuHAK**
* Added MSYS/MINGW32 toolchain
* Fixed Github Action to upload artifacts even if some toolchains failed to compile
* Fixed some scripts to work with non-default toolchains
* Changed generated toolchain naming scheme

### CODE REVIEW

File: .github/workflows/compilation.yml [hunk 1]
```diff
@@ -8,53 +8,72 @@ on:
...
     - name: Runs all the stages in the shell
+      continue-on-error: true
       run: |
         export PS2DEV=$PWD/ps2dev
         export PS2SDK=$PS2DEV/ps2sdk
         export GSKIT=$PS2DEV/gsKit
         export PATH="/usr/local/opt/gnu-sed/libexec/gnubin:$PATH"
         export PATH=$PATH:$PS2DEV/bin:$PS2DEV/ee/bin:$PS2DEV/iop/bin:$PS2DEV/dvp/bin:$PS2SDK/bin
-        ./build-all.sh
+        # ./build-all.sh
```
Comment (AKuHAK) [inline .github/workflows/compilation.yml:58]: This can be changed when ps2client will be fixed

Comment (fjtrujy) [reply .github/workflows/compilation.yml:58]: I think that is cleaner to modify `006-ps2client-extra.sh` and put there the condition `if MINGW .... exit`

Comment (AKuHAK) [reply .github/workflows/compilation.yml:58]: Its possible, but is it so crucial to have ps2client inside the toolchain?

Comment (AKuHAK) [reply .github/workflows/compilation.yml]: I managed to fix ps2client

**Comment by AKuHAK** [comment]
Before merging this PR, it is useful to merge ps2client PR: https://github.com/ps2dev/ps2client/pull/13

**Comment by fjtrujy** [comment]
Nice @AKuHAK !
Can you just remove from the description the:
`Temporarily removed ps2client from toolchains, cause it doesn't compile on windows`
Because now it is not happening?
Thanks

**Pull request merged**